src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -44,37 +39,83 @@ public MovieSession add(MovieSession movieSession) {
-            Predicate allConditions = criteriaBuilder.and(moviePredicate, datePredicate);
-            criteriaQuery.select(root).where(allConditions);
-            root.fetch("movie");
-            root.fetch("cinemaHall");
-            return session.createQuery(criteriaQuery).getResultList();
+            return Optional.ofNullable(session.get(MovieSession.class, id));
         } catch (Exception e) {
-            throw new DataProcessingException("Can't get available sessions for movie with id: "
-                    + movieId + " for date: " + date, e);
+            throw new DataProcessingException("Can't get a movieSession by id: " + id, e);
         }
     }
 
     @Override
-    public Optional<MovieSession> get(Long id) {
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
         try (Session session = HibernateUtil.getSessionFactory().openSession()) {
-            Query<MovieSession> query = session.createQuery("FROM MovieSession ms "
-                    + "JOIN FETCH ms.movie m "
-                    + "JOIN FETCH ms.cinemaHall ch "
-                    + "WHERE ms.id = :id", MovieSession.class);
-            query.setParameter("id", id);
-            return query.uniqueResultOptional();
+            LocalDateTime start = date.atTime(0,0,0);
+            LocalDateTime end = date.atTime(23, 59, 59);
+            Query<MovieSession> movieSessionQuery
+                    = session.createQuery("from MovieSession ms "
+                            + "where ms.movie.id = :movieId "
+                            + "and ms.showTime > :start "
+                            + "and ms.showTime < :end",
+                    MovieSession.class);
+            movieSessionQuery.setParameter("movieId", movieId);
+            movieSessionQuery.setParameter("start", start);
+            movieSessionQuery.setParameter("end", end);
+            return movieSessionQuery.getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException(
+                    "Can't get available sessions on the scheduled date: " + date, e);
+        }
+    }
+
+    @Override
+    public boolean update(MovieSession movieSession) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        Transaction transaction = null;
+
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.update(movieSession);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("updating " + movieSession + " failed", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+
+        return get(movieSession.getId()).get().equals(movieSession);
```

Comment:
The update method returns a boolean by comparing the updated entity with the one retrieved from the database. This approach might not be reliable if the entity has fields that are not persisted or if the database automatically modifies some fields (e.g., timestamps). Consider returning a boolean based on the success of the update operation itself, rather than comparing objects.